File: LuciansLusciousLasagna.cs
class Lasagna
{
    public int ExpectedMinutesInOven()
    {
        return 40;
    }

    public int RemainingMinutesInOven(int time)
    {
        return 40 - time;
    }

    public int PreparationTimeInMinutes(int layers)
    {
        return layers * 2;
    }

    public int ElapsedTimeInMinutes(int layers, int time)
    {
        int result = PreparationTimeInMinutes(layers) + time;
        return result;
    }
}

File: fsharp/LuciansLusciousLasagna.fs
module LuciansLusciousLasagna

let expectedMinutesInOven = 40

let remainingMinutesInOven timeInOven = expectedMinutesInOven - timeInOven

let preparationTimeInMinutes layers = layers * 2

let elapsedTimeInMinutes layers minutes =
    preparationTimeInMinutes layers + minutes
